google-cloud-apis/src/com/google/cloud/tools/intellij/cloudapis/DefaultCloudApiUiPresenter.java

```diff
@@ -45,15 +48,16 @@ public void updateCloudLibraryVersionLabel(@Nullable String text, @Nullable Icon
   }
 
   /**
-   * Inits the presenter, creates all extension points and adds necessary event listeners and
-   * handlers.
+   * Inits the presenter when add libraries dialog is created, creates all extension points and adds
+   * necessary event listeners and handlers.
    */
-  void init(GoogleCloudApiSelectorPanel cloudApiSelectorPanel) {
+  void init(@NotNull GoogleCloudApiSelectorPanel cloudApiSelectorPanel) {
     this.cloudApiSelectorPanel = cloudApiSelectorPanel;
 
     cloudApiUiExtensions = CloudApiUiExtension.EP_NAME.getExtensions();
     for (CloudApiUiExtension uiExtension : cloudApiUiExtensions) {
-      uiExtension.init(this);
+      // TODO: will be implemented in the next PR.
+      Collection<JComponent> customComponents = uiExtension.createCustomUiComponents();
```

Comment:
changes look great.

just curious what the strategy with this one will be: how will these custom components be places in the UI? Would it be better to just present a single custom JPanel to extend the base UI (which is similar to how a lot of internal idea UI stuff works)?

Reply:
I am worried about alignment of module combobox with BOM combo box in this case a bit. I was planning to add components to existing grid spots below the module/project selector space, but if possible, let's change to single panel - much more flexible. I will experiment with layouts to see how this will work out.

Reply:
now that I think about it, aren't there two custom areas for injecting cutsom ui components: at the top area near the GCP project selector, in the library details panel (next to the library description, version, etc)? I guess we should only support the use-case we immediately need and expand later if/when needed.

Reply:
actually we do need to support both immediately I think: at the top area (due to the BOM selector), in the library details panel (due to the maven coordinate info, and others).

Reply:
Exactly, great point, for now we will just need components in the line below module/project selection (BOM), and following these steps can expand later to as many places as we need.